### PR TITLE
[feat] Add Stripe Handling

### PR DESCRIPTION
**Pull request by aquamatthias**
# Description

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `nox`

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://some.engineering/code-of-conduct).

### CODE REVIEW

File: fixbackend/domain_events/events.py [hunk 1]
```diff
@@ -230,16 +230,16 @@ class ProductTierChanged(Event):
...
     This event is emitted when a user gets an AWS Marketplace subscription.
     """
 
-    kind: ClassVar[str] = "aws_marketplace_subscription_created"
-
+    kind: ClassVar[str] = "subscription_created"
     workspace_id: Optional[WorkspaceId]
     user_id: UserId
     subscription_id: SubscriptionId
+    method: Literal["aws_marketplace", "stripe"]
```
Comment (meln1k) [inline fixbackend/domain_events/events.py:242]: we should probably use an StrEnum here.

File: fixbackend/ids.py [hunk 2]
```diff
@@ -33,6 +35,7 @@ class NotificationProvider(StrEnum):
     opsgenie = "opsgenie"
 
 
+BillingPeriod = Literal["month", "day"]
```
Comment (meln1k) [inline fixbackend/ids.py:38]: StrEnum?

File: fixbackend/subscription/aws_marketplace.py [hunk 3]
```diff
@@ -209,12 +209,12 @@ async def create_billing_entry(
                 usage = int(len(summaries) * month_factor)
                 on_payment_free_tier = product_tier.paid is False
                 if on_payment_free_tier or usage == 0:
-                    log.info(f"AWS Marketplace: customer {customer} has no usage")
+                    log.info(f"{kind}: subscription {subscription.id} has no usage")
```
Comment (meln1k) [inline fixbackend/subscription/aws_marketplace.py:212]: Let's add an easy to search prefix here: I often query `AWS Marketplace` in loki to investigate problems.

File: fixbackend/subscription/models.py [hunk 4]
```diff
@@ -52,4 +70,6 @@ class BillingEntry:
 
 
 # Multiple payment methods are possible, but for now we only support AWS Marketplace
-SubscriptionMethod = Union[AwsMarketplaceSubscription]
+SubscriptionMethod = Union[AwsMarketplaceSubscription, StripeSubscription]
+
+SubscriptionMethodType = TypeVar("SubscriptionMethodType", bound=SubscriptionMethod)
```
Comment (meln1k) [inline fixbackend/subscription/models.py:75]: Do we need this generic here? isn't union above enough?

**Review by meln1k — APPROVED** [review]
A couple of nits, otherwise LGTM

File: fixbackend/subscription/models.py (hunk 4 above)
Comment (aquamatthias) [reply fixbackend/subscription/models.py:75]: No: when I give you a StripeSubscription, I also want to receive one - not a generic payment method.

File: fixbackend/ids.py (hunk 2 above)
Comment (aquamatthias) [reply fixbackend/ids.py:38]: Defer to follow up PR.

File: fixbackend/domain_events/events.py (hunk 1 above)
Comment (aquamatthias) [reply fixbackend/domain_events/events.py:242]: Defer to follow up PR.

**Pull request merged**